### PR TITLE
config: health checker timeout and interval should be greater than 0

### LINKED ISSUE #3729: [health checker] Validate that required fields are greater than 0
*Description*:
While protobuf validation exists for things like [interval and timeout](https://github.com/envoyproxy/envoy/blob/84cacd2a5598f8b7d50fe0ccd3ef8a6deecfc2fc/api/envoy/api/v2/core/health_check.proto#L27) in cluster health checking configuration, there is [no check 
for whether values passed in make sense](https://github.com/envoyproxy/envoy/blob/84cacd2a5598f8b7d50fe0ccd3ef8a6deecfc2fc/source/common/upstream/health_checker_base_impl.cc#L19). Setting the interval to something nonsensical like 0 makes the process spin in an error loop so would be a lot more straightforward to have envoy fail fast in cases like these.

### PR DESCRIPTION
**Pull request by dio**
*Description*:
This patch adds value validation for interval and timeout fields to accept values greater than 0.

*Risk Level*: Low
*Testing*: Unit, manual by setting timeout or interval to 0s
*Docs Changes*: N/A
*Release Notes*: N/A

Fixes #3729 

Signed-off-by: Dhi Aurrahman <dio@rockybars.com>

### CODE REVIEW

File: api/envoy/api/v2/core/health_check.proto [hunk 1]
```diff
@@ -21,10 +21,16 @@ option (gogoproto.equal_all) = true;
 message HealthCheck {
   // The time to wait for a health check response. If the timeout is reached the
   // health check attempt will be considered a failure.
-  google.protobuf.Duration timeout = 1 [(validate.rules).duration.required = true];
+  google.protobuf.Duration timeout = 1 [
+    (validate.rules).duration = {required: true, gt: {}},
```
Comment (mattklein123) [inline api/envoy/api/v2/core/health_check.proto]: nit: IMO saying greater than 0 is a bit more clear here. Same below.

Comment (dio) [reply api/envoy/api/v2/core/health_check.proto]: I thought the same, but then I found that the others use `gt: {}` only. E.g. https://github.com/envoyproxy/envoy/blob/5d2bd6e0f16167db1f57282525384d0efb140aff/api/envoy/config/filter/http/buffer/v2/buffer.proto#L23

**Comment by dio** [comment]
@mattklein123 do you think I can just add config tests inside the `health_checker_impl_test` and redis' `config_test`? Or is it a good idea to introduce a new file containing config tests for the health check config?

**Comment by mattklein123** [comment]
@dio I would just add tests wherever is easiest. (Obviously just make sure the validator actually runs.) Thank you!

**Comment by mattklein123** [comment]
@junr03 please take a pass on this once tests are added. Thank you!

**Comment by dio** [comment]
Test is added. PTAL. Thanks!

File: api/envoy/api/v2/core/health_check.proto (hunk 1 above)
Comment (junr03) [reply api/envoy/api/v2/core/health_check.proto]: mmm I still think that having the explicit value is more clear. I would rather add it here and update the other protos in a separate PR for explicitness.

Comment (dio) [reply api/envoy/api/v2/core/health_check.proto]: @mattklein123 @junr03 will do. I also think that we should automatically render that requirement into doc, e.g. REQUIRED, GREATER THAN 0 SEC. But I believe that should be down in a different PR. Related: https://github.com/envoyproxy/envoy/issues/3731

Comment (dio) [reply api/envoy/api/v2/core/health_check.proto]: @junr03 done in 1d2b066.

**Pull request merged**